lib/chef/resource/windows_task.rb

```diff
@@ -227,9 +227,18 @@ def validate_user_and_password(user, password)
         end
       end
 
+      def user_is_system?(user)
+        if Chef::ReservedNames::Win32::Security::SID.from_account(user) == Chef::ReservedNames::Win32::Security::SID.LocalSystem ||
```

Comment:
I haven't played with this in pry... so it may not work at all, but you might be able to clean this up a little with something like:
```
account = Chef::ReservedNames::Win32::Security::SID.from_account(user)
account == Chef::ReservedNames::Win32::Security::SID.LocalSystem
```
I am not sure if you can call the class method on the object or not, but if you can it might also work as:

```
account = Chef::ReservedNames::Win32::Security::SID.from_account(user)
account.LocalSystem || account.NtLocal || account.NtNetwork || account.BuiltinUsers
```
Again, I haven't tried this myself.... but in my head it compiles. :man_shrugging: :wink:

Reply:
@rhass What do you think about this change?